### ISSUE TITLE
webmail/cpanel "subdomains"

### ISSUE THREAD

**Issue by brinsleyparagao** [engintron/engintron#207]
Hello

In my server when try access on to this address;
cpanel.domain.com or webmail.domain.com the screen no render fine, thats show a corrupted  mal formed html panels.

so how to fix this?

**Comment by fevangelou** [comment]
The subdomains "webmail" and "cpanel" are being proxied by Apache if I recall properly to the internal cPanel ports used.

The solution another user mentioned was to comment out the line "proxy_set_header X-Forwarded-Host $host;" in proxy_params_common.

Do it and let me know if it works. If it does, I'll include custom rules to correct this behaviour.

**Comment by brinsleyparagao** [comment]
Hello my friend....
No work to me, in two servers...
I try reinstall the engitron in stable and later in mainline version, but the error persist

**Comment by fevangelou** [comment]
If you re-install, you will lose the change in the code I just wrote you 
about.

Regards,

Fotis Evangelou

Engintron is Nginx on cPanel, done right!
Learn more at: https://engintron.com

**Comment by brinsleyparagao** [comment]
I know,
i try comment that line:
# proxy_set_header X-Forwarded-Host $host;

But dont work, when dont work, i try reinstall to try fix. but without succes

**Comment by brinsleyparagao** [comment]
Some another tip to fix this ?

**Comment by RedDot-3ND7355** [comment]
Worked out fine for me!
Are you using cloudflare?

**Comment by brinsleyparagao** [comment]
@RedDot-3ND7355 no, no work to me, i no use CF and anyone CDN... :(

**Comment by hostfusion** [comment]
Hi,

same issue, i have commented the `>proxy_set_header X-Forwarded-Host $host;` but don´t fix the issue.

Regards

PS

**Comment by hostfusion** [comment]
Hi again,

confirmed that this issue is with engintrone, if i disable engintron the subdomains for webmail and cpanel work again fine. Comment the line >proxy_set_header X-Forwarded-Host $host; not fix the issue.

I hope this information help @fevangelou 

Regards

PS

**Comment by vmunich** [comment]
Hi,

Using Engintron 1.6.2, commenting `proxy_set_header X-Forwarded-Host $host;` in `proxy_params_common` did not work. I had to disable Engintron ):

**Comment by jcmdln** [comment]
In my current near-stock deployment of Engintron, ```cpanel.domain.com``` works without issue. You may want to review your DNS Zone file to ensure that this record is created, try rebuilding it, or confirm that the nameservers you are using have a valid IP to your server (only valid if using Google, OpenDNS, or nameservers you do not have write access to).

You may want to try whitelisting the IP altogether using the steps outlined for fixing issues with CloudFlare via the wiki, though this shouldn't be needed with custom nameservers (provided your DNS is properly clustered).

**Comment by marklajer** [comment]
We have the same issue

**Comment by fevangelou** [comment]
Those with subdomains not working, can you please provide a couple of
actual URLs for me to check remotely?

**Comment by hostfusion** [comment]
Hi @fevangelou the issue is not with the url, its when you login for example in webmail.domain.com or cpanel.domain.com, are missconfigured with engintron active, like the screenshots
![captura de pantalla 2016-07-30 a las 12 00 42](https://cloud.githubusercontent.com/assets/6918330/17269617/a56cc7a4-564d-11e6-9ce8-f27c71517697.png)
![captura de pantalla 2016-07-30 a las 12 01 09](https://cloud.githubusercontent.com/assets/6918330/17269621/b23b1800-564d-11e6-9b97-c8c9007714d6.png)

Regards

PS

**Comment by fevangelou** [comment]
That's why I asked for a domain to check. I obviously meant a
cpanel.something.tld subdomain.

**Comment by hostfusion** [comment]
Hi, for example adcare.es, alicantepress.com, limtegrales.com , the second domain is pointing to a external ip but the email is running in the server.

Regards

PS

**Comment by shivsingh7150** [comment]
I also have same issue i am using cloud linux and after updating it not working any way, Comment the line >proxy_set_header X-Forwarded-Host $host; not fix the issue.

Please help

**Comment by fevangelou** [comment]
@hostfusion The problem as far as I can see is that some static files (CSS or/and JS) are not properly loaded because of the proxying taking place. Since I don't currently have a server with webmail.* and cpanel.* setup, could you please copy/paste the HTML code (in a GitHub Gist or similar) of the page you land after you login e.g. in webmail.*?

**Comment by shivsingh7150** [comment]
Thanks for reply, after upgrading to latest release now its working by port login like

http://domainname.com:2095 or http://domainname.com/webmail

but still not working on webmail.domainname.com

**Comment by brinsleyparagao** [comment]
Hello, I make a update to new version but the bug with cpanel/webmail subdmain persists
`http://imgur.com/mGtGR9y`
And i try comment the string but the error its here !
`proxy_set_header X-Forwarded-Host $host;`

**Comment by fevangelou** [comment]
@brinsleyparagao Since I don't currently have a server with webmail.* and cpanel.* setup, could you please copy/paste the HTML source code (in a GitHub Gist or similar) of the page you land after you login e.g. in webmail.*?

**Comment by brinsleyparagao** [comment]
@fevangelou if you need a cpanel server i can create one account to you test, if accept of course. I (have skype, hangout and discord chats to very fast communications - if accept a account to tests)

`
<!DOCTYPE html>

<html lang="en" dir="ltr">
    <head>
        <title>Webmail - Main</title>
        <meta charset="utf-8" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1">

        
        <meta name="theme-color" content="#293a4a">

		    
        <meta name="mobile-web-app-capable" content="yes">
        <meta name="mobile-web-app-status-bar-style" content="default">

        
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="default">
        


        <link rel="shortcut icon" href="/brand/favicon.ico?84864307" type="image/x-icon" />

        <link rel="stylesheet" type="text/css" href="/cPanel_magic_revision_1470255310/webmail/paper_lantern/_assets/css/master-ltr.cmb.min.css" />

        <link rel="stylesheet" type="text/css" href="/cPanel_magic_revision_1470255310/webmail/paper_lantern/mail_clients/mail_clients.css" />

            <script type="text/javascript" src="/cPanel_magic_revision_1470255310/webmail/paper_lantern/_assets/js/master-legacy.cmb.min.js"></script>
        

              <script type="text/javascript" src="/cPanel_magic_revision_1470255309/cjt/cpanel-all-min.js?locale=en"></script>
        <script type="text/javascript">
            CPANEL.validate.hide_validation_summary = true;
        </script>
        
        <script>
            (function(){
                window.PAGE = window.PAGE || {};
                window.NVData = {};
                
                
                window.NVData["hideUserPreferencesOverlay"] = "1";
                
                window.thisTheme = "paper_lantern";
            })();
        </script>

        
                <link rel="stylesheet" type="text/css" href="/styled/current_style/styles.css?84864307" />
                
                <link rel="stylesheet" type="text/css" href="/styled/current_style/sprites/icon_spritemap.css?84864307" />
        

    </head>
    <body id="cpanel_body" class="yui-skin-sam">
        <div id="wrap">
            <header id="masterAppContainer">
                <div class="navbar navbar-inverse navbar-cpanel navbar-fixed-top" role="navigation">
                    <div class="navbar-header">
                        
                            <a id="lnkHeaderHome" class="navbar-brand" target="_top" href="./index.html">
                                <img id="imgLogo" class="navbar-brand-logo" src="/brand/webmail.png?84864307" alt="" title=""  />
                            </a>
                        
                        <div class="navbar-preferences">
                            <span id="userPreferencesContainer" class="navbar-overlay-wrapper">
<div class="btn-group">
    <button id="btnUserPref" data-toggle="dropdown" class="btn btn-primary dropdown-toggle btn-sm">
        <span id="userImg" class="glyphicon glyphicon-user"></span>
        <span id="lblUserNameTxt" class="hidden-inline-xs">eu@brinsley.top</span>
        <span id="caretImg" class="caret"></span>
    </button>
    <ul class="dropdown-menu dropdown-menu-right">

        
            
            <li class="mail-app">
                <a id="lnkUserPrefroundcube" href="./index.html?mailclient=roundcube" class="app-name">
                    Roundcube
                </a>
                <a id="lnkUserPrefroundcubeFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="roundcube" title="Set as my default webmail application">
                    <span id="imgUserPrefroundcube" class="fa fa-star fa-fw pull-right" data-default-webmail-app="roundcube" alt="Set as my default webmail application"></span>
                </a>
            </li>
            
        <li class="divider"></li>
        
        <li>
            <a id="lnkUserPrefChangePwd" href="./mail/change_password.html" class="app-name">Password & Security            </a>
            <a id="lnkUserPrefChangePwdFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="change_password" title="Set as my default application">
                <span id="imgUserPrefChangePwd" class="fa fa-star fa-fw pull-right" data-default-webmail-app="change_password" alt="Set as my default application"></span>
            </a>
        </li>
        <li>
            <a id="lnkUserPrefConfigMailClient" href="./mail/clientconf.html" class="app-name">Configure Mail Client            </a>
            <a id="lnkUserPrefConfigMailClientFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="configure_mail_client" title="Set as my default application">
                <span id="imgUserPrefConfigMailClient" class="fa fa-star fa-fw pull-right" data-default-webmail-app="configure_mail_client" alt="Set as my default application"></span>
            </a>
        </li>

        
        <li>
            <a id="lnkUserPrefConfigCalendarContactClient" href="./mail/calendars_and_contacts/index.html.tt" class="app-name">Calendars and Contacts            </a>
            <a id="lnkUserPrefConfigCalendarContactClientFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="calendars_and_contacts" title="Set as my default application">
                <span id="imgUserPrefConfigCalendarContactClient" class="fa fa-star fa-fw pull-right" data-default-webmail-app="calendars_and_contacts" alt="Set as my default application"></span>
            </a>
        </li>
        
        <li>
            <a id="lnkUserPrefEmailFiltering" href="./mail/filters/userfilters.html" class="app-name">Email Filters            </a>
            <a id="lnkUserPrefEmailFilteringFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="email_filters" title="Set as my default application">
                <span id="imgUserPrefEmailFiltering" class="fa fa-star fa-fw pull-right" data-default-webmail-app="email_filters" alt="Set as my default application"></span>
            </a>
        </li>
                <li>
            <a id="lnkUserPrefForwarder" href="./mail/fwds.html" class="app-name">Forwarders            </a>
            <a id="lnkUserPrefForwarderFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="forwarders" title="Set as my default application">
                <span id="imgUserPrefForwarder" class="fa fa-star fa-fw pull-right" data-default-webmail-app="forwarders" alt="Set as my default application"></span>
            </a>
        </li>
                <li>
            <a id="lnkUserPrefAutoResponders" href="./mail/autores.html" class="app-name">Autoresponders            </a>
            <a id="lnkUserPrefAutoRespondersFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="autoresponders" title="Set as my default application">
                <span id="imgUserPrefAutoResponders" class="fa fa-star fa-fw pull-right" data-default-webmail-app="autoresponders" alt="Set as my default application"></span>
            </a>
        </li>
        
        <li>
            <a id="lnkUserPrefUpdateContactInfo" href="./contact/index.html" class="app-name">Contact Information            </a>
            <a id="lnkUserPrefUpdateContactInfoFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="contact_information" title="Set as my default application">
                <span id="imgUserPrefContactInfo" class="fa fa-star fa-fw pull-right" data-default-webmail-app="contact_information" alt="Set as my default application"></span>
            </a>
        </li>
        
        <li>
            <a id="lnkUserPrefEmailTrace" href="./mail/route.html" class="app-name">Track Delivery            </a>
            <a id="lnkUserPrefEmailTraceFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="track_delivery" title="Set as my default application">
                <span id="imgUserPrefEmailTrace" class="fa fa-star fa-fw pull-right" data-default-webmail-app="track_delivery" alt="Set as my default application"></span>
            </a>
        </li>        <li class="divider"></li>
        <li>
            <a id="lnkUserPrefBoxTrapper" href="./mail/manage.html" class="app-name">BoxTrapper            </a>
            <a id="lnkUserPrefBoxTrapperFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="boxtrapper" title="Set as my default application">
                <span id="imgUserPrefBoxTrapper" class="fa fa-star fa-fw pull-right" data-default-webmail-app="boxtrapper" alt="Set as my default application"></span>
            </a>
        </li>

        
        
        
        
        
        
        
        
        
        
        
        
        <li class="sub-menu-item">
            <a id="lnkboxtrapper_review_queue" href="./mail/showq.html" class="app-name">Review Queue            </a>
            <a id="lnkboxtrapper_review_queueFav" href="javascript:void(0);" class="app-fav" data-default-webmail-app="boxtrapper_review_queue" title="Set as my default application">
                <span id="imgboxtrapper_review_queue" class="fa fa-star fa-fw pull-right" data-default-webmail-app="boxtrapper_review_queue" alt="Set as my default application"></span>
            </a>
        </li>
        
        
    </ul>
</div>
                                <div class="navbar-overlay-message row hidden">
                                    <div class="col-xs-10">
                                        <h4 id="txtNavbarOverlayMessage">Change your email configuration options anytime.</h4>
                                        <button id="btnNavbarOverlayConfirm" type="button" class="btn">Got it</button>
                                    </div>
                                    <div class="col-xs-2 text-center">
                                        <span id="icnNavbarOverlayIndicator" class="glyphicon glyphicon-user fa-3x"></span>
                                    </div>
                                </div>
                                <span id="logoutContainer" class="navbar-overlay-wrapper">
                                   <div class="btn-group">
                                    <a target="_top" href="/logout/?locale=en">
                                      <button id="lnkHeaderLogout" class="btn logout-btn btn-sm">
                                        <span id="logoutImg" class="glyphicon glyphicon-log-out"></span>
                                        <span id="lblLogout" class="hidden-inline-xs">Logout</span>
                                      </button>
                                    </a>
                                   </div>
                                </span>
                            </span>
                        </div>
                    </div>
                </div>
            </header>
            
            
            <div id="content" class="container-fluid">
            
            
                <iframe id="mailFrame" class="mail_client_roundcube"  src="/cpsess5511271966/3rdparty/roundcube/index.php" seamless="seamless" width="100%" height="950px"></iframe>
                
                <style>

                    #content {
                        width: 100%;
                        margin: 0;
                        padding: 0;
                    }

                    /**
                     * We need to use position absolute here because safari
                     * on iOS 9 does not support position fixed for iframes.
                     *
                     * margin-top is used here instead of padding-top
                     * because of the smartmobile view in Horde which
                     * is using jquery-mobile. jquery-mobile counts the padding
                     * as part of the height and will cause the footer to appear
                     * in the wrong spot.
                     *
                     * The height needs to be caculated at 100vh - whatever we
                     * are using for the margin. Setting the height to just 100vh
                     * causes the last email in the list to be partially hidden
                     * when using the jquery-mobile view particularly on iOS.
                     * Adjusting the height to account for the margin will
                     * correctly display the last item fully.
                     */
                    #mailFrame {
                        border: 0px;
                        height: calc(100vh - 50px);
                        margin-top: 50px;
                        position: absolute;
                        top: 0;
                        width: 100%;
                        bottom: 0;
                    }

                    div#wrap {
                        background-repeat: repeat-x;
                    }

                    footer {
                        display: none;
                    }

                    header .navbar {
                        margin: 0px;
                        border: 0px;
                    }

                    @media (max-width: 667px) {

                        #mailFrame {
                            height: calc(100vh - 30px);
                            margin-top: 30px;
                        }

                    }

                </style>
            </div>
            
            <footer>
                <div class="container-fluid">
                    <div class="navbar">
                        <ul class="nav navbar-nav">
                            <li>
                                <a id="lnkFooterHome" href="./index.html">Home                                </a>
                            </li>
                            <li>
                                <a id="lnkFooterTrademark" href="./trademarks.html" target="_blank">Trademarks                                </a>
                            </li>
                            <li>
                                <a id="lnkFooterDocs" href="https://go.cpanel.net/paperlanterndocs" target="_blank">Documentation                                </a>
                            </li>
                        </ul>

                        <div class="navbar-brand" style="display:inline-block; z-index:2147483647; visibility:visible;">
                            <a id="lnkPoweredByCpanel" href="http://www.cpanel.net" target="cpanel" title="cPanel, Inc." style="display:inline-block; z-index:2147483647; visibility:visible;">
                                <img id="imgPoweredByCpanel" src="/cPanel_magic_revision_1470255310/img-sys/powered_by_cpanel.png" alt="cPanel, Inc." style="display:inline-block; z-index:2147483647; visibility:visible;">
                            </a>
                            <sub id="txtCpanelVersion" style="display:inline-block; z-index:2147483647; visibility:visible;">58.0.20</sub>
                        </div>
                    </div>
                </div>
            </footer>
        </div>

        <script>
            CPANEL.lang.Close=CPANEL.lang.close="Close";CPANEL.lang.cancel="Cancel";CPANEL.lang.unlimited="Unlimited";CPANEL.lang.or="or";CPANEL.lang.ajax_loading="Loading …";CPANEL.lang.password_generator="Password Generator";CPANEL.lang.generate_password="Generate Password";CPANEL.lang.advanced="Advanced Options";CPANEL.lang.confirm_copy_password="I have copied this password in a safe place.";CPANEL.lang.use_password="Use Password";CPANEL.lang.length="Length";CPANEL.lang.alpha_characters="Alpha Characters";CPANEL.lang.nonalpha_characters="Non Alpha Characters";CPANEL.lang.both="Both";CPANEL.lang.lowercase="Lowercase";CPANEL.lang.uppercase="Uppercase";CPANEL.lang.numbers="Numbers";CPANEL.lang.symbols="Symbols";CPANEL.lang.password_strength="Password Strength";CPANEL.lang.passwords_match="Passwords Match";CPANEL.lang.password_validator_no_spaces="Password cannot have spaces.";CPANEL.lang.password_validator_no_empty="Password cannot be empty.";CPANEL.lang.password_validator_no_match="Passwords do not match.";CPANEL.lang.Validation_Errors="Validation Errors";CPANEL.lang.strength_phrase_very_weak="Very Weak";CPANEL.lang.strength_phrase_weak="Weak";CPANEL.lang.strength_phrase_ok="OK";CPANEL.lang.strength_phrase_strong="Strong";CPANEL.lang.strength_phrase_very_strong="Very Strong";CPANEL.lang.click_to_close="Click to close.";CPANEL.lang.toggle_more="more »";CPANEL.lang.toggle_less="less »";CPANEL.lang.json_error=CPANEL.lang.Error="Error";CPANEL.lang.json_parse_failed="JSON parse failed.";CPANEL.lang.ajax_error="AJAX Error";CPANEL.lang.ajax_try_again="Please refresh the page and try again.";
        </script>
        
        <script type="text/javascript" src="/cPanel_magic_revision_1470255310/webmail/paper_lantern/_assets/base.js"></script>

<script>
$(function(){

    var mail_client;

           mail_client = "roundcube"
    
    $("#mailFrame").load(function(){
        var mailFrameContents;
        mailFrameContents = $("#mailFrame").contents();
        if(mail_client === "roundcube"){
            mailFrameContents.find(".button-logout").hide();
            mailFrameContents.find(".about-link").hide();
            mailFrameContents.find(".username").hide();
            mailFrameContents.find("#topline").hide();
        }

        if(mail_client === "horde"){
            mailFrameContents.find("#horde-logout").hide();
            mailFrameContents.find(".smartmobile-logout").hide();
        }

        if(mail_client === "sqmail"){
            hide_squrrel_mail_logout(mailFrameContents);
        }

    });

    function hide_squrrel_mail_logout(mailFrameContents) {
        var squrrel_mail_frame = mailFrameContents.find('frame[name="right"]')[0];
        if(squrrel_mail_frame){
            $(squrrel_mail_frame.contentDocument.getElementsByClassName("sqm_headerSignout")).hide();

            mailFrameContents.find('frame[name="right"]')[0].onload = function() {
               $(this.contentDocument.getElementsByClassName("sqm_headerSignout")).hide()
            };
        }

    }
}

);
</script>		<script type="text/javascript">
		if (("standalone" in window.navigator) && window.navigator.standalone) {
		    jQuery(document).ready(function($) {
		        $("a").click(function(event) {
		            var alink = document.createElement('a');
		            alink.href = $(this).attr("href");
		            if (alink && alink.protocol.indexOf('http') !== -1 && alink.host.indexOf(document.location.host) !== -1) {
		                event.preventDefault();
		                window.location = alink.href;
		            }
		        });
		    });
		}
		</script>
        <script>
        $(
            function() {
                var img = $("#imgPoweredByCpanel");
                if(img === null || img.length <= 0 ||
                    img.attr('src') !== "/cPanel_magic_revision_1470255310/img-sys/powered_by_cpanel.png") {
                    window.location="/";
                }

                // Warning Duplicate IDs
                var ids = {};
                $('[id]').each(function(){
                    if(ids[this.id]) {
                        console.warn('Duplicate ID #'+this.id);
                    }
                    ids[this.id] = 1;
                });
            }
        );
        </script>
    </body>
</html>
`

**Comment by fevangelou** [comment]
@brinsleyparagao I can't understand why cPanel does that. You'll have to raise this issue with cPanel support as they know better how/if there is any other proxying performed when you can webmail.domain.tld. If they provide you with insight to adapt code in Engintron, I will happily do it.

**Comment by fevangelou** [comment]
@brinsleyparagao BTW, I think cPanel (under Tweak Settings) has an option to force redirect any webmail.domain.tld to the hostname of the server. You might want to enable this until the issue is resolved.

**Comment by brinsleyparagao** [comment]
@fevangelou i try disable cpanel's subdomain proxed - and works !
I understand too, because in v1.61 work fine. but if make this work perfectly.

`http://imgur.com/NKkKDs9`

**Comment by fevangelou** [comment]
Engintron introduced no change that could cause this issue after v1.6.1. Perhaps it was an internal change in cPanel (?).

**Comment by expssh** [comment]
Hi, Fotis. Even if disable Proxy subdomains it is not working. There is really serious issue. Which log's or page source you need to check, tell me I will provide.

**Comment by mexkiv** [comment]
I added this in my proxy_params_common

```
# force ssl
if ($scheme = "http") {
    return 301 https://$host$request_uri;
}
```

**Comment by DanielRivera1309** [comment]
Hello,

I want to revive this issue. apparently the issue is the apache port. cPanel is looking for acces via new apache port (8080) however, one reason can be the firewall that is blocking it.

I just open the port and the issue has been solve.

Now, the question is: Can I add a nginx rule or cPanel configuration for avoid it? because with the port 8080 open in the firewall, we are exposed to DDoS atack directly to apache.

(PD: be patience with me, is my first day using engintron in 5 servers)

**Comment by fevangelou** [comment]
You can restrict access for internal use only. See the related post in the Engintron Wiki.

**Comment by DanielRivera1309** [comment]
I know @fevangelou howerver if we close the 8080 port in CSF, cpanel subdomains doesn't works

**Comment by fevangelou** [comment]
I'm referring to this: https://github.com/engintron/engintron/wiki/Restrict-access-to-ports-8080-&-8443-used-by-Apache-only-for-Nginx

**Comment by nermindadanovic** [comment]
Hello,
for the last couple of days I've installed Engintron on my dedicated server, made it work with Cloudflare for 130 websites, most of them are on Cloudflare.
But I can't manage to make webmail subdomain to work, and all my customers use webmail.domain.com, and cpanel.domain.com
Is there a solution for webmail.domain.com since I have 130 websites with couple of thousands of emails, being used daily. 
I cant make them use port or domain.com/webmail. Somehow Roundcube breaks apart, CSS JS, session etc.

Is there a solution to make it work? Because If I can't make it work then I have to look for another solution, and Engintron is awesome for my needs, except this issue with webmail.domain.com.

Can I exclude webmail subdomain from caching somehow?

I've tried:
if ($scheme = "http") {
    set $redirToSSL "yes";
}
if ($host ~* "^webmail\.") {
    set $isWebmail "please";
}
set $webmailRedirection $redirToSSL$isWebmail;
if ($webmailRedirection = "yesplease") {
    return 301 https://$host$request_uri;
}
if ($host ~* "^(webmail|cpanel|whm|webdisk|cpcalendars|cpcontacts)\.") {
    set $PROXY_FORWARDED_HOST '';
}
commenting proxy_set_header X-Forwarded-Host $host;
